st-exerciseviewer/src/main/java/de/saring/exerciseviewer/parser/impl/TimexPwxParser.java

```diff
@@ -654,7 +654,13 @@ private EVExercise parseWorkoutSamples(EVExercise exercise, Node workoutNode) {
                     } else if (childName.equals("torq")) {
                         // Not implemented in ExerciseSample class
                     } else if (childName.equals("cad")) {
-                        sample.setCadence(Short.valueOf(sampleChildren.item(j).getTextContent()));
+                        final Float cadence = Float.valueOf(sampleChildren.item(j).getTextContent());
```

Comment:
When the textContent of the item is null, then will Float.valueOf throw an exception. It would be better to check whether the textContent is not null and not emty.

Reply:
This code change was just a suggestion, it does not fix fully the import anyway. I would rather, before attempting to parse this file - just validate the schema, and only proceed if the validation passes. This will also help give a more useful message to the user.